### ISSUE TITLE
CamelJdbcTest leaks a statement

### ISSUE THREAD

**Issue by zhfeng** [apache/camel-quarkus#4752]
When running the `testMoveDataBetweenDatasources`, the agroal detects some resources leak issues. I think it could be a potenial issue in the upstream `camel-jdbc` component. 

```
2023-04-06 21:31:01,717 WARN  [io.agr.pool] (executor-thread-1) Datasource 'camel-ds': JDBC resources leaked: 0 ResultSet(s) and 1 Statement(s)
2023-04-06 21:31:01,718 WARN  [io.agr.pool] (executor-thread-1) Datasource 'camel-ds': JDBC resources leaked: 0 ResultSet(s) and 1 Statement(s)
2023-04-06 21:31:01,719 WARN  [io.agr.pool] (executor-thread-1) Datasource 'camel-ds': JDBC resources leaked: 0 ResultSet(s) and 1 Statement(s)
```

**Comment by zhfeng** [comment]
I open a ticket https://issues.apache.org/jira/browse/CAMEL-19225

**Comment by jamesnetherton** [comment]
Looks to be fixed after upgrade to Camel 3.0.0-M3.
